cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManager.java

```diff
@@ -591,4 +643,35 @@ public String toString() {
       return CACHED_VALUES;
     }
   }
+
+  /**
+   * {@link _allUuidToUserTaskInfoMap} stores active and completed user tasks for history. Some historic user requests are
+   * less important for the user, e.g. {@link EndPoint.USER_TASKS} is not very interesting. To prevent un-interesting but
+   * frequently-called endpoints from flooding out interesting requests, we need to periodically cleanup requests. We can set
+   * per endpoint limit and utilize the {@link UserTaskScanner} to do the cleaning.
+   */
+  private class UserTaskCleanerInfo {
+    private final int _limit;
+    private final LinkedList<UUID> _tasks;
```

Comment:
Why would we prefer `LinkedList` over `LinkedHashMap`?

Reply:
For the current design, we just need a FIFO data structure, so `LinkedHashMap` is not necessary. We could use it for other designs.

Reply:
Ah sorry -- I meant `LinkedHashSet`?

Reply:
We just need FIFO, not sure we need property in Set